src/ldp/http/SparqlUpdateBodyParser.ts

```diff
@@ -30,7 +31,7 @@ export class SparqlUpdateBodyParser extends BodyParser {
     // It is impossible to check if object mode is enabled in Node 10 (without accessing private variables)
     const options = { objectMode: request.readableObjectMode };
     const toAlgebraStream = pipeSafely(request, new PassThrough(options));
-    const dataCopy = pipeSafely(request, new PassThrough(options));
+    const dataCopy = guardStream(pipeSafely(request, new PassThrough(options)));
```

Comment:
Should `pipeSafely` already guard?

Optimization: if `pipeSafely` receive a guarded stream as input, it can unwrap it, pipe it, and then guard the result.

Reply:
Didn't add guards to `pipeSafely` indeed because of what happens when a stream is already guarded. I'm wondering now if we can check if a stream is already guarded using instanceof, and if not then guard the stream. (Could also be used to make sure we don't guard a stream twice). Otherwise I would have to add an `unGuard` function that removes all the listeners that are potentially attached.

Reply:
Maybe you want to _actually_ add the nominal symbol to the stream. Or maybe you want to turn it into a `guarded = true` so we can just check for that field. Or indeed directly the `unguard` function as a signal of presence.